src/data/monsters.txt

```diff
@@ -376,23 +376,23 @@ Knob Goblin Mutant	88	kg_mutant.gif	Atk: 53 Def: 47 HP: 43 Init: 70 Meat: 55 P:
 Knob Goblin poseur	533	kg_poseur.gif	Atk: 4 Def: 3 HP: 5 Init: 50 Meat: 20 P: goblin	dope gangsta bling-bling (10)	out-of-tune biwa (10)	terrible poem (20)
 Knob Goblin Sous Chef	76	kg_souschef.gif	Atk: 20 Def: 18 HP: 15 Init: 60 Meat: 20 P: goblin	Knob Goblin spatula (7)	Knob sausage (15)	tomato (15)	wad of dough (15)
 Knob Goblin Very Mad Scientist	86	kg_verymadsci.gif	Atk: 45 Def: 40 HP: 30 Init: 70 Meat: 45 P: goblin	Knob Goblin melon baller (5)	Knob Goblin seltzer (15)	Knob Goblin superseltzer (15)	scrumptious reagent (25)	bunch of square grapes (c19)	smudged alchemical recipe (p5)	Cobb's Knob Menagerie key (c0)	Subject 37 file (c0)
-Knott Slanding	617	slanding.gif	NOCOPY ULTRARARE NOMANUEL Atk: 150 Def: 135 HP: 150 Init: 10000 E: cold Meat: 970-1342 P: beast	Dallas Dynasty Falcon Crest shield (100)
-Knott Yeti	99	yeti.gif	Atk: 105 Def: 92 HP: 90 Init: 60 E: cold Meat: 200 P: beast	yeti fur (30)
+Knott Slanding	617	slanding.gif	NOCOPY NOMANUEL ULTRARARE Atk: 150 Def: 135 HP: 150 Init: 10000 Meat: 1156 P: beast E: cold	Dallas Dynasty Falcon Crest shield (100)
+Knott Yeti	99	yeti.gif	Atk: 105 Def: 92 HP: 90 Init: 60 Meat: 200 P: beast E: cold	yeti fur (30)
 Koopa Troopa	116	koopa.gif	Atk: 20 Def: 18 HP: 15 Init: 60 P: beast	black pixel (50)	green pixel (80)	white pixel (50)
-L imp	124	limp.gif	Atk: 44 Def: 39 HP: 44 Init: 60 E: hot Meat: 30 P: demon	Imp Ale (30)	cast (30)	flaming crutch (30)
+L imp	124	limp.gif	Atk: 44 Def: 39 HP: 44 Init: 60 Meat: 30 P: demon E: hot	Imp Ale (30)	cast (30)	flaming crutch (30)
 Lamz0r N00b	140	n00b.gif	Atk: 79 Def: 71 HP: 69 Init: 70 Meat: 30 P: dude	33398 scroll (30)	draggin' ball hat (10)	plastic guitar (10)	Nardz energy beverage (p30)
 large kobold	224	lower_k.gif	Atk: 57 Def: 49 HP: 60 Init: 45 P: humanoid	ring of aggravate monster (15)	small box (40)	kobold kibble (c0)
-Larry of the Field of Signs	984	larrysignfield.gif	NOCOPY BOSS Atk: 70 Def: 63 HP: 70 Init: 50 P: demon	observational glasses (100)
-larval filthworm	477	filthworm1.gif	NOCOPY Atk: 165 Def: 148 HP: 180 Init: 50 Meat: 50 E: stench P: bug	filthworm hatchling scent gland (10)
+Larry of the Field of Signs	984	larrysignfield.gif	BOSS NOCOPY Atk: 70 Def: 63 HP: 70 Init: 50 P: demon	observational glasses (100)
+larval filthworm	477	filthworm1.gif	NOCOPY Atk: 165 Def: 148 HP: 180 Init: 50 Meat: 50 P: bug E: stench	filthworm hatchling scent gland (10)
 lemon-in-the-box	109	linbox.gif	Atk: 17 Def: 15 HP: 12 Init: 75 P: construct	box (20)	lemon (50)
 Lesser Fruit Golem	284	lfruitgol.gif	Atk: 17 Def: 15 HP: 9 Init: 50 P: construct	grapefruit (15)	grapes (15)	lemon (15)	orange (15)	strawberry (15)
-lihc	32	lich.gif	Atk: 20 Def: 18 HP: 11 Init: 60 ED: spooky EA: spooky EA: sleaze Meat: 36 P: undead	lihc eye (30)
-Little Man in the Canoe	540	canoeman.gif	Atk: 165 Def: 148 HP: 150 Init: 70 E: sleaze P: constellation	line (30)	star (30)	star (30)
+lihc	32	lich.gif	Atk: 20 Def: 18 HP: 11 Init: 60 Meat: 36 P: undead E: sleaze	lihc eye (30)
```

Comment:
A lihc has four attacks, three of which are spooky elemental and one of which is sleaze. The script has observed that both EA and ED were present, and collapsed them into E, but picked the wrong EA.

To clarify, as I understand EA, ED, and E:
* ED: this monster is element X
* EA: this monster has at least one attack with damage X
* E: this monster has element X, and at least one elemental attack with damage X

Notably, EA can occur multiple times, as here.

Reply:
In my workspace, I have restored the original monsters.txt (and the obsolete parsing of poison) so I can work on this.

Looking at the code before I started this work:
The MonsterData.java:
```
  private Element attackElement;
  private Element defenseElement;
```
There is only only attack element and one defense element.

MonsterDatabase.java
```
        } else if (option.equals("E:")) {
          if (tokens.hasMoreTokens()) {
            value = tokens.nextToken();
            Element element = MonsterDatabase.parseElement(value);
            if (element != Element.NONE) {
              attackElement = element;
              defenseElement = element;
            }
          }
          continue;
        } else if (option.equals("ED:")) {
          if (tokens.hasMoreTokens()) {
            value = tokens.nextToken();
            Element element = MonsterDatabase.parseElement(value);
            if (element != Element.NONE) {
              defenseElement = element;
            }
          }
          continue;
        } else if (option.equals("EA:")) {
          if (tokens.hasMoreTokens()) {
            value = tokens.nextToken();
            Element element = MonsterDatabase.parseElement(value);
            if (element != Element.NONE) {
              attackElement = element;
            }
          }
          continue;
        }
```
I think your understanding of the attributes is correct.
Notice that since we do not currently support multiple elemental attacks, we simply store the last one found.
So, yeah - my collapsing of EA+ED to E should check that both EA and ED are the same element.
Unfortunately, since we do not store additional EA, I can't get both without doing something about "we only store one EA".

I'll ponder this. I wonder how many more monsters there are with more than one EA?
I can debug that by logging when we parse the strings from monsters.txt. :)

Reply:
Some monsters, like Lord Spookyraven, can attack with any element.

Reply:
I have fixed the handling for collapsing EA and ED into E.
I have left multiple EA attributes as is: only keep the last one listed.
However, I made it log when it sees this - only the lihc, at the moment - so we will be reminded to maybe think more about this. :)